build.gradle.kts

```diff
@@ -29,7 +29,7 @@ repositories {
     mavenCentral()
 }
 
-val appVersion = layout.projectDirectory.file("release/version.txt").asFile.readText().trim()
+val appVersion = releaseVersion()
```

Comment:
We have 4 different version variables in this script that _I think_ are all the same thing. 

`appVersion`, `releaseVersion`, `projectVersion`, and the normal `version`. 😵 

In the other scripts we just have `releaseVersion` which gets assigned to `version`.

Is it possible to do the same thing here? I realize you are just following the same patterns that already existed in this build script, but with the addition of the `releaseVersion()` function it might be a good time to consider cleaning it up.

Reply:
I can investigate - `projectVersion` has a comment saying it is required for configuration cache compatibility. I wasn't exactly sure why that was the case, but I'll look into that and into consolidating those variables.

Reply:
After looking into this, unless you know a way around it, the `projectVersion` variable is unavoidable (though maybe it could use a better name).

From the docs [here](https://docs.gradle.org/current/userguide/configuration_cache.html#config_cache:not_yet_implemented:accessing_top_level_at_execution)...
```
Build scripts written in Kotlin cannot store tasks that reference top-level methods or variables at execution time in the configuration cache at all. This limitation exists because the captured script object references cannot be serialized.
```

Reply:
Tyler and I paired on this to come up with a solution. Now there is only `releaseVersion` and the regular `project.version` which is just like our other repositories.